modules/assim.batch/R/pda.utils.R

```diff
@@ -608,12 +608,23 @@ pda.adjust.jumps.bs <- function(settings, jcov, accept.count, params.recent) {
 ##'
 ##' @author Ryan Kelly, Istem Fer
 ##' @export
-pda.generate.knots <- function(n.knot, n.param.all, prior.ind, prior.fn, pname) {
+pda.generate.knots <- function(n.knot, sf, probs.sf, n.param.all, prior.ind, prior.fn, pname) {
   # By default, all parameters will be fixed at their median
   probs <- matrix(0.5, nrow = n.knot, ncol = n.param.all)
   
-  # Fill in parameters to be sampled with probabilities sampled in a LHC design
-  probs[, prior.ind] <- PEcAn.emulator::lhc(t(matrix(0:1, ncol = length(prior.ind), nrow = 2)), n.knot)
+  # if the params are going to be scaled leave them out, if sf is NULL nothing happens
+  inds <- prior.ind[!prior.ind %in% which(pname %in% sf)]
+  
+  if(length(inds) !=0){
+    # Fill in parameters to be sampled with probabilities sampled in a LHC design
+    probs[, inds] <- PEcAn.emulator::lhc(t(matrix(0:1, ncol = length(inds), nrow = 2)), n.knot)
+  }
+  
+  inds <- prior.ind[prior.ind %in% which(pname %in% sf)]
+  
+  if(!is.null(sf) & length(inds) !=0){
```

Comment:
Single `&` is a vectorized conditional. Double `&&` is an evaluation left to right. Not sure it technically matters here, but for cases like this, you probably want `&&`.